src/statement.rs

```diff
@@ -93,7 +93,8 @@ impl<'con, 'param, 'col, S, A> Statement<'con, 'param, 'col, S, A> {
         mut self,
         parameter_number: SQLUSMALLINT,
         parameter_type: DataType,
-        value: Option<&'p T>,
+        value: &'p T,
+        indicator: Option<&'p SQLLEN>,
```

Comment:
It will probably be hard to make this compile-time without constant expression generics, but I think we could add a runtime check against `CDataType::buffer_len`? The main question I see is whether this should just panicor be returned as a failure.  Personally, I would prefer to just `assert!` here due to this usually being a logic rather than a runtime error, like indexing into a slice out of its bounds.

Reply:
(Of course, one could also go ahead and introduce `SqlNullable<T>` and `SqlString<A>` and such types to avoid exposing bare indicators at all, but I think this would be more appropriate for `odbc-rs` than for `odbc-safe`.)

Reply:
I like your reasoning. Let's check against `buffer_len` and `panic!` if it fails.

Reply:
Added the necessary assertion. Since the indicator reference lifetimes have the same bounds as the parameters, this should also prohibit them changing their value between parameter binding and statement execution.